styx-cli/src/main/java/com/spotify/styx/cli/CliMain.java

```diff
@@ -274,7 +275,23 @@ private void run(CliContext cliContext) {
     } catch (ExecutionException e) {
       final Throwable cause = e.getCause();
       if (cause instanceof ApiErrorException) {
-        cliOutput.printError("API error: " + cause.getMessage());
+        final ApiErrorException apiError = (ApiErrorException) cause;
+        if (apiError.getCode() == UNAUTHORIZED.code()) {
+          if (!apiError.isAuthenticated()) {
+            cliOutput.printError(
+                "API error: Unauthorized: Please set up Application Default Credentials or set the "
+                    + "GOOGLE_APPLICATION_CREDENTIALS env var to point to a file defining the "
+                    + "credentials.\n"
+                    + "\n"
+                    + "Hint: Try setting up credentials using gcloud: \n"
+                    + "\n"
+                    + "\t$ gcloud auth application-default login");
+          } else {
+            cliOutput.printError("API error: Unauthorized");
```

Comment:
At this moment, this should never happen, right? Just asking.

Reply:
Will happen if the service rejects the credentials. Which it will start doing for e.g. external credentials at some point. And the cli should should behave well wrt that (even "old" ones).

Reply:
cool